src/librustc_mir/transform/acs_propagate.rs

```diff
@@ -40,25 +42,49 @@ use pretty;
 
 #[derive(PartialEq, Debug, Eq, Clone)]
 enum Either<'tcx> {
-    Top,
     Lvalue(Lvalue<'tcx>),
     Const(Constant<'tcx>),
 }
 
-impl<'tcx> Lattice for Either<'tcx> {
-    fn bottom() -> Self { unimplemented!() }
+#[derive(Debug, Clone)]
+enum AcsLattice<'tcx> {
+    Bottom,
```

Comment:
Also, I realized after I wrote that code that `WBottom` existed and that it would have been better to use that. However, I'm still planning to remove the requirement that lattices have `bottom`, again inspired by https://ghc.haskell.org/trac/ghc/wiki/Hoopl/Cleanup, and so I didn't bother to switch to a cleaner version.